libs/iwinfo/Makefile

```diff
@@ -1,121 +0,0 @@
-#
```

Comment:
If you are removing this, can you make sure nodewatcher agent works with latest `iwinfo`?

Reply:
No need for it since everything that was patched in was merged over 2 years ago.

Reply:
I see no reason why it wouldnt,it was only extended from 3 years ago.
I have not seen breaking changes in it.
Also,in order to test it,I need to be able to deploy builders and generate images on working Nodewatcher

Reply:
https://git.openwrt.org/?p=project/iwinfo.git;a=commit;h=16597a70bc946937130db9f0832e3d0c448c8233

Commit by Kostko

Reply:
OK, so if we get this merged, you will then make sure nodewatcher agents continues to work?

Reply:
I can test it in my Nodewatcher, but monitoring does not work there.
So there's really no point.

Can I build the imagebuilder so you can add it to test channel onto Nodewatcher?

Reply:
What do you mean? You can locally configure a node to connect to your local instance of the nodewatcher?

Reply:
You can and should be able to test this locally. No need to deploy it to the whole network for a developer to test it. Once you test it locally, then we can deploy it so that we can test it more in the wild, before it becomes a default.

Reply:
I have Nodewatcher running on one of our servers to test stuff out.
Only monitoring is not configured there.

I cant manually generate imagebuilder and image for one device to test out if it nodewatcher-agent sends data

Reply:
Hm. So when I am developing this, I have nodewatcher running on my own machine. Then I generate an image with it, flash the device, and have them both in the same LAN. And then I configure manually nodewatcher to connect to that device and try to fetch data. So even without Babel. Why you could not do the same?

Reply:
I havent thought of that.
Will give it a try today or tommorow

Reply:
Can you tell me what configure in settings.py?

Reply:
I think this is the diff:

```diff
diff --git a/nodewatcher/settings.py b/nodewatcher/settings.py
index e3082e1a..0dd31e04 100644
--- a/nodewatcher/settings.py
+++ b/nodewatcher/settings.py
@@ -317,8 +318,8 @@ INSTALLED_APPS += [
     'nodewatcher.modules.services.dhcp',
     'nodewatcher.modules.services.nodeupgrade',
     'nodewatcher.modules.services.watchdog',
-    'nodewatcher.modules.routing.olsr',
-    'nodewatcher.modules.routing.babel',
+    #'nodewatcher.modules.routing.olsr',
+    #'nodewatcher.modules.routing.babel',
     'nodewatcher.modules.authentication.public_key',
     'nodewatcher.modules.vpn.tunneldigger',
     'nodewatcher.modules.events.sinks.db_sink',
@@ -329,7 +330,7 @@ INSTALLED_APPS += [
     'nodewatcher.modules.frontend.mynodes',
     'nodewatcher.modules.frontend.setup',
     'nodewatcher.modules.frontend.statistics',
-    'nodewatcher.modules.frontend.topology',
+    #'nodewatcher.modules.frontend.topology',
     'nodewatcher.modules.frontend.generator',
     'nodewatcher.modules.frontend.map',
     'nodewatcher.modules.administration.banner',
@@ -482,46 +488,44 @@ TELEMETRY_PROCESSOR_PIPELINE = (
 
 MONITOR_RUNS = {
     'latency': {
-        'workers': 10,
-        'interval': 600,
+        'workers': 5,
+        'interval': 60,
         'processors': (
-            'nodewatcher.modules.routing.olsr.processors.GlobalTopology',
-            'nodewatcher.modules.routing.babel.processors.IncludeRoutableNodes',
+            'nodewatcher.core.monitor.processors.GetAllNodes',
             'nodewatcher.modules.monitor.measurements.rtt.processors.RttMeasurement',
             'nodewatcher.modules.monitor.datastream.processors.TrackRegistryModels',
             'nodewatcher.modules.monitor.measurements.rtt.processors.StoreNode',
+            # Validators should start here in order to obtain previous state.
+            'nodewatcher.modules.monitor.validation.reboot.processors.RebootValidator',
+            'nodewatcher.modules.monitor.validation.version.processors.VersionValidator',
+            'nodewatcher.modules.monitor.validation.interfaces.processors.InterfaceValidator',
+            # Telemetry processors should be below this point.
+            'nodewatcher.modules.monitor.sources.http.processors.HTTPTelemetry',
+            'nodewatcher.modules.monitor.http.general.processors.GeneralInfo',
+            'nodewatcher.core.monitor.processors.GeneralInfo',
+            'nodewatcher.modules.monitor.http.resources.processors.SystemStatus',
+            'nodewatcher.modules.monitor.http.interfaces.processors.DatastreamInterfaces',
+            'nodewatcher.modules.monitor.http.clients.processors.ClientInfo',
+            'nodewatcher.modules.monitor.http.survey.processors.SurveyInfo',
+            'nodewatcher.modules.sensors.generic.processors.GenericSensors',
+            'nodewatcher.modules.vpn.tunneldigger.processors.DatastreamTunneldigger',
             'nodewatcher.modules.administration.status.processors.NodeStatus',
             'nodewatcher.modules.monitor.datastream.processors.NodeDatastream',
-        ),
-    },
-
-    'telemetry': {
-        'workers': 15,
-        'interval': 300,
-        'max_tasks_per_child': 50,
-        'processors': (
-            'nodewatcher.core.monitor.processors.GetAllNodes',
-            'nodewatcher.modules.routing.olsr.processors.GlobalTopology',
-            'nodewatcher.modules.routing.babel.processors.IncludeRoutableNodes',
-            'nodewatcher.modules.monitor.datastream.processors.TrackRegistryModels',
-            'nodewatcher.modules.routing.olsr.processors.NodeTopology',
-            TELEMETRY_PROCESSOR_PIPELINE,
             'nodewatcher.modules.monitor.datastream.processors.MaintenanceBackprocess',
-            'nodewatcher.modules.administration.status.processors.PushNodeStatus',
         ),
     },
 
-    'telemetry-push': {
-        # This run does not define any scheduling or worker information, so it will only be
-        # executed on demand.
-        'processors': (
-            'nodewatcher.modules.monitor.unknown_nodes.processors.DiscoverUnknownNodes',
-            'nodewatcher.modules.monitor.sources.http.processors.HTTPGetPushedNode',
-            'nodewatcher.modules.identity.base.processors.VerifyNodeIdentity',
-            'nodewatcher.modules.monitor.datastream.processors.TrackRegistryModels',
-            TELEMETRY_PROCESSOR_PIPELINE,
-        ),
-    },
+    #'telemetry-push': {
+    #    # This run does not define any scheduling or worker information, so it will only be
+    #    # executed on demand.
+    #    'processors': (
+    #        'nodewatcher.modules.monitor.unknown_nodes.processors.DiscoverUnknownNodes',
+    #        'nodewatcher.modules.monitor.sources.http.processors.HTTPGetPushedNode',
+    #        'nodewatcher.modules.identity.base.processors.VerifyNodeIdentity',
+    #        'nodewatcher.modules.monitor.datastream.processors.TrackRegistryModels',
+    #        TELEMETRY_PROCESSOR_PIPELINE,
+    #    ),
+    #},
 
     'datastream': {
         'workers': 2,
@@ -532,16 +536,16 @@ MONITOR_RUNS = {
         ),
     },
 
-    'topology': {
-        'workers': 5,
-        'interval': 60,
-        'processors': (
-            'nodewatcher.modules.routing.olsr.processors.GlobalTopology',
-            'nodewatcher.modules.routing.olsr.processors.NodeTopology',
-            'nodewatcher.modules.monitor.topology.processors.Topology',
-            'nodewatcher.modules.monitor.datastream.processors.NetworkDatastream',
-        ),
-    },
+    # 'topology': {
+    #     'workers': 3,
+    #     'interval': 60,
+    #     'processors': (
+    #         #'nodewatcher.modules.routing.olsr.processors.GlobalTopology',
+    #         #'nodewatcher.modules.routing.olsr.processors.NodeTopology',
+    #         'nodewatcher.modules.monitor.topology.processors.Topology',
+    #         'nodewatcher.modules.monitor.datastream.processors.NetworkDatastream',
+    #     ),
+    # },
 }
 
 # Identifier of the run that should be used to handle HTTP pushes.
@@ -571,12 +575,14 @@ DATASTREAM_BACKEND_SETTINGS = {
     },
 }
 
-OLSRD_MONITOR_HOST = '127.0.0.1'
-OLSRD_MONITOR_PORT = 2006
+#OLSRD_MONITOR_HOST = '10.254.128.33'
+#OLSRD_MONITOR_PORT = 2006
 
 # UUID of the node that is performing measurements (usually the node where the nodewatcher
 # monitor is running on).
-MEASUREMENT_SOURCE_NODE = ''
+#MEASUREMENT_SOURCE_NODE = ''
+#MEASUREMENT_SOURCE_NODE = '64840ad9-aac1-4494-b4d1-9de5d8cbedd9'
+MEASUREMENT_SOURCE_NODE = 'c5c397e8-b9e4-4865-8e17-22f433c17c13'
 
 # Storage for generated firmware images.
 GENERATOR_STORAGE = 'django.core.files.storage.FileSystemStorage'
```

So this should make nodewatcher try to pull monitoring data from all registered nodes, no matter the routing protocol status. Otherwise we try to fetch data only from nodes visible by the routing daemon.

Now the only tricky thing is that you have to get your nodewatcher running inside Docker to be able to connect to the node. But that should be true if node is in the same LAN as your development computer.

I am not 100% sure if this is all that is currently needed, but this is what I had at some point if I see correctly.